File: java-ver/src/main/java/com/prvz/url_checker/exception/IllegalTaskOperationException.java
package com.prvz.url_checker.exception;


import lombok.Getter;
import org.springframework.util.StringUtils;

@Getter
public class IllegalTaskOperationException extends RuntimeException {

    private static final Long serialVersionUID = 1L;
    private final String taskId;

    public IllegalTaskOperationException(String message, String taskId) {
        super(StringUtils.hasText(message) ?
                ("Некорректная операция с задачей: " + taskId) : message);
        this.taskId = taskId;
    }
}

File: src/main/kotlin/com/prvz/url_checker/exception/IllegalTaskOperationException.kt
package com.prvz.url_checker.exception

open class IllegalTaskOperationException(val taskId: String, message: String? = null) :
        RuntimeException(message ?: "Некорректная операция с задачей: $taskId") {
    companion object {
        const val serialVersionUID = 1L
    }
}